src/data_structures.rs

```diff
@@ -104,11 +105,12 @@ pub trait PCProof: Clone + ark_ff::ToBytes + CanonicalSerialize + CanonicalDeser
 
 /// A proof of satisfaction of linear combinations.
 #[derive(Clone, CanonicalSerialize, CanonicalDeserialize)]
-pub struct BatchLCProof<F: Field, P: Polynomial<F>, PC: PolynomialCommitment<F, P>> {
+pub struct BatchLCProof<F: Field, P: Polynomial<F>, PC: PolynomialCommitment<F, P, S>, S: FieldBasedCryptographicSponge<F>> {
```

Comment:
It doesn't have to be a `FieldBasedCryptographicSponge`; it can be any sponge. Also, the field of the sponge does not necessarily match the field of the polynomials. For example, in curve-based recursion, it usually doesn't.

Reply:
Oh, that makes sense. Will change the API soon.